app/models/manageiq/providers/cloud_manager/server_group.rb

```diff
@@ -0,0 +1,66 @@
+class ManageIQ::Providers::CloudManager::ServerGroup < ::ServerGroup
+  has_many :vms, dependent => :nullify
+
+  def self.class_by_ems(ext_management_system)
+    ext_management_system&.class_by_ems(:ServerGroup)
+  end
+
+  def self.create_server_group_queue(ems_id, options = {})
+    task_opts = {
+      :action => "creating Server Group for cloud id #{cloud_id}",
+      :ems_id => ems_id
+    }
+
+    queue_opts = {
+      :class_name  => 'ManageIQ::Providers::CloudManager::ServerGroup',
+      :method_name => 'create_server_group',
+      :role        => 'ems_operations',
+      :queue_name  => ext_management_system.queue_name_for_ems_operations,
```

Comment:
Do you have `ext_management_system` set here?  You need to do `ext_management_system = ExtManagementSystem.find(ems_id)` above

Reply:
If you pass `ext_management_system` in then you don't need to do the query here which is probably why this code worked where it was lifted from